src/routes.tsx

```diff
@@ -41,13 +50,79 @@ import { ProposalDetailsPage } from './app/pages/ProposalDetailsPage'
 import { ConsensusBlocksPage } from './app/pages/ConsensusBlocksPage'
 import { ConsensusAccountsPage } from './app/pages/ConsensusAccountsPage'
 import { ConsensusTransactionsPage } from './app/pages/ConsensusTransactionsPage'
+import { FC, useEffect } from 'react'
 
 const NetworkSpecificPart = () => (
   <ThemeByNetwork network={useRequiredScopeParam().network}>
     <Outlet />
   </ThemeByNetwork>
 )
 
+/**
+ * Use the passed RouteObject unless we have a fixed layer.
+ *
+ * Normally, this will just wrap the RouteObject into an array.
+ * Unless the condition is true, because in that case, and empty array will be returned.
+ */
+const unlessFixedLayer = (route: RouteObject): RouteObject[] => (fixedLayer ? [] : [route])
+
+/**
+ * Use the passed RouteObject unless we have a fixed network and layer.
+ *
+ * Normally, this will just wrap the RouteObject into an array.
+ * Unless the condition is true, because in that case, and empty array will be returned.
+ */
+const unlessFixedNetworkAndLayer = (route: RouteObject): RouteObject[] =>
+  hasFixedNetworkAndLayer ? [] : [route]
+
+/**
+ * Use the passed RouteObject unless we are fixed on a ParaTime.
+ *
+ * Normally, this will just wrap the RouteObject into an array.
+ * Unless the condition is true, because in that case, and empty array will be returned.
+ */
+const unlessFixedOnParatime = (route: RouteObject): RouteObject[] => (isFixedOnParatime ? [] : [route])
+
+/**
+ * Use the passed RouteObject unless we are fixed on consensus.
+ *
+ * Normally, this will just wrap the RouteObject into an array.
+ * Unless the condition is true, because in that case, and empty array will be returned.
+ */
+const unlessFixedOnConsensus = (route: RouteObject): RouteObject[] => (isFixedOnConsensus ? [] : [route])
+
+/**
+ * Use the passed RouteObject if we have a fixed layer but no network
+ *
+ * Normally, this will just wrap the RouteObject into an array.
+ * Unless the condition is true, because in that case, and empty array will be returned.
+ */
+const ifFixedLayerButNoNetwork = (route: RouteObject): RouteObject[] =>
+  !!fixedLayer && !fixedNetwork ? [route] : []
+
+/**
+ * In case of being restricted to a specific layer, jump to a network
+ *
+ * This should be rendered on the landing page,
+ * in order to redirect to a network, since we don't want the
+ * opening graph.
+ */
+const RedirectToNetwork: FC = () => {
+  const navigate = useNavigate()
+  const layer = fixedLayer
```

Comment:
This declaration is kind of useless.

Reply:
Yeah it comes from the age when `fixedLayer()` was still a function. Now removed.